File: tests/data/program_analysis/strings/Translation/str01.py
from strings import *

def str01():
    str1 = String(10)
    str2 = String(10)
    str3 = String(15)

    str1.set_value("abcdef")
    str2.set_value("ijklmnop")
    str3.set_value(str1 + str2)

    print(f'str1: len = {len(str1)}, value = "{str1._val}"')
    print(f'str2: len = {len(str2)}, value = "{str2._val}"')
    print(f'str1//str2: len = {len(str1+str2)}, value = "{str1+str2}"')
    print(f'str3: len = {len(str3)}, value = "{str3._val}"')

str01()

File: tests/data/program_analysis/strings/str01.f
C     Strings and string concatenation

      program str01
      character(len = 10) str1, str2, str3*15

      str1 = "abcdef"
      str2 = "ijklmnop"

      str3 = str1 // str2       ! concatenation

      write (*, 10) "str1", len(str1), str1
      write (*, 10) "str2", len(str2), str2
      write (*, 10) "str1//str2", len(str1//str2), str1//str2
      write (*, 10) "str3", len(str3), str3
 10   format(A, ': len = ', I2, '; value = "', A, '"')

      stop
      end program str01
      
